mediatek: backport support for KTD2026/7 rgb(w) led controller

goldwrt opened the pull request:
The patch will add the Linux kernel mainline driver "leds-ktd202x" for the
KinetIC KTD2026 and KTD2027 RGB/RBGW controller with I2C interface that was
introduced in kernel version 6.7, last changed in mainline on 2024-05-31.

At least the Acer Connect Vero W6m (a variant of the Acer Predator Connect
W6 without 2.5G eth1 port, usb3 port, and the 6 on-board gpio RGB LEDs) is
equipped with a KTD2026, and maybe other router devices also are.

As the RGB LED driven by a KTD2026 is the W6m's only LED (designed and used
by the stock firmware as status LED), the driver should be ready to be
loaded early at boot time.

Signed-off-by: George Oldfort <openwrt@10099.de>

musashino205 commented:
Please,

- remove the merge commits and squash commits to single one
- backport the upstream commits for that controller to mediatek target instead of adding source files
  - https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/log/drivers/leds/rgb/?qt=grep&q=ktd202x
- remove unnecessary `Subject: ` prefix from the commit title

hauke commented:
you can push your changes back into this branch and then reopen this PR.

hauke commented:
Please move the patches to `target/linux/generic/backport-6.6/` and then add the entry to build the driver to: [package/kernel/linux/modules/leds.mk](https://github.com/openwrt/openwrt/blob/main/package/kernel/linux/modules/leds.mk)

goldwrt committed:
generic: backport support for KTD2026/7 rgb(w) led controller

This commit adds the Linux kernel mainline driver "leds-ktd202x" for the
KinetIC KTD2026 and KTD2027 RGB/RBGW controller with I2C interface that was
introduced in kernel version 6.7, last changed in mainline on 2024-05-31.

At least the Acer Connect Vero W6m (a variant of the Acer Predator Connect
W6 without 2.5G eth1 port, usb3 port, and the 6 on-board gpio RGB LEDs) is
equipped with a KTD2026 (and a single RGB LED attached to it used by the
stock firmware as status LED), and maybe other router devices also are.

Signed-off-by: George Oldfort <openwrt@10099.de>
Link: https://github.com/openwrt/openwrt/pull/16860
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

goldwrt committed:
mediatek: filogic: add KTD2026/7 driver to Acer Connect Vero W6m

This commit adds kmod-leds-ktd202x to the OpenWrt image for the device
"Acer Connect Vero W6m" which is equipped with one KTD2026 controlling the
device's status LED via I2C.

Signed-off-by: George Oldfort <openwrt@10099.de>
Link: https://github.com/openwrt/openwrt/pull/16860
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
